src/vm/gcenv.ee.cpp

```diff
@@ -861,13 +861,19 @@ void GCToEEInterface::StompWriteBarrier(WriteBarrierParameters* args)
 #endif
 
 #ifdef FEATURE_USE_SOFTWARE_WRITE_WATCH_FOR_GC_HEAP
-        if (args->write_watch_table != nullptr)
+        if (g_sw_ww_enabled_for_gc_heap && (args->write_watch_table != nullptr))
         {
             assert(args->is_runtime_suspended);
             g_sw_ww_table = args->write_watch_table;
         }
 #endif // FEATURE_USE_SOFTWARE_WRITE_WATCH_FOR_GC_HEAP
 
+#if defined(_ARM64_)
+        g_lowest_address = args->lowest_address;
+        g_highest_address = args->highest_address;
+
+        ::StompWriteBarrierResize(args->is_runtime_suspended, args->requires_upper_bounds_check);
+#else
```

Comment:
Arm64 needs these variables updated before calling ::StompWriteBarrierResize().  Since arm64 is also forced to `SuspendEE()`/`RestartEE()` the `FlushProcessWriteBuffers()` is not needed